src/txn/txn.c

```diff
@@ -1159,9 +1159,15 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
             continue;
         }
 
-        /* Ignore the already resolved updates. */
-        if (upd->prepare_state == WT_PREPARE_RESOLVED)
+        /*
+         * Performing an update on the same key where the truncate operation is performed can lead
+         * to updates that are already resolved in the updated list. Ignore the already resolved
+         * updates.
+         */
+        if (upd->prepare_state == WT_PREPARE_RESOLVED) {
+            WT_ASSERT(session, upd->type == WT_UPDATE_TOMBSTONE);
```

Comment:
I don't understand why it can't be a standard update. I think it is possible to have two prepared standard update from the same transaction.

Reply:
There shouldn’t be any resolved prepared updates in the update list until it is a delete update created for a truncate operation. If it is a normal update/delete, it must not be resolved already when the transaction is committing now.